pkg/traceutil/trace.go

```diff
@@ -167,6 +195,17 @@ func (t *Trace) logInfo(threshold time.Duration) (string, []zap.Field) {
 		zap.Duration("duration", totalDuration),
 		zap.Time("start", t.startTime),
 		zap.Time("end", endTime),
-		zap.Strings("steps", steps)}
+		zap.Strings("steps", steps),
+		zap.Int("step_count", len(t.steps))}
 	return msg, fs
 }
+
+func (t *Trace) hasField(f Field) bool {
+	for i, v := range t.fields {
+		if v.Key == f.Key {
+			t.fields[i].Value = f.Value
```

Comment:
if we update this trace field value here, the func naming needs to be changed. hasField should be a func name for a read only operation.

Reply:
probably use: updateFieldIfExist instead.

Reply:
Fixed. Thanks for pointing out.